Gems/ROS2/Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -91,6 +91,7 @@ namespace ROS2
     void ROS2SpawnerComponent::SpawnEntity(const SpawnEntityRequest request, SpawnEntityResponse response)
     {
         AZStd::string spawnableName(request->name.c_str());
+        AZStd::string spawnableNamespace(request->robot_namespace.c_str());
```

Comment:
RosifyName could be used to sanitize external input.

Reply:
Since the request was to spawn with a specific name / namespace, there could be a problem, especially with automated runs, if the actual name/namespace was different. Thus, I believe it is better to validate (using our utils) instead and return failure if the validation does not pass.